### PR TITLE
v0.82.0 - Star rating modifier

### PR DESCRIPTION
**Pull request by sonnyprince**
- Addition of extraSmall ratings modifier.
- Amend listing style to work with new rating size

## UI Review Checks

Before:
screen shot 2018-09-10 at 20 59 12

After:
screen shot 2018-09-10 at 20 49 58

- [x] This PR has been checked with regard to our brand guidelines
- [x] UI Documentation has been [updated]

(PR for that here: https://github.com/justeat/global-component-library/pull/53)

- [x] This code has been checked with regard to our accessibility standards

## Browsers Tested

- [x] Chrome
- [x] Edge
- [x] Internet Explorer 11
- [x] Mobile (i.e. iPhone/Android - please list device)

### CODE REVIEW

File: src/scss/components/_ratings.scss [hunk 1]
```diff
@@ -13,6 +14,11 @@ $star-spacing: $star-count;
     width: $star-size-small * $star-count + $star-spacing;
 }
 
+.c-rating--extraSmall {
+    height: $star-size-extraSmall;
+    width: $star-size-extraSmall * $star-count + ($star-spacing * 1);
```
Comment (DamianMullins) [inline src/scss/components/_ratings.scss]: Do you need to do `$star-spacing * 1`?

Comment (sonnyprince) [reply src/scss/components/_ratings.scss]: Fixed!

File: src/scss/components/_ratings.scss [hunk 2]
```diff
@@ -13,6 +14,11 @@ $star-spacing: $star-count;
     width: $star-size-small * $star-count + $star-spacing;
 }
 
+.c-rating--extraSmall {
+    height: $star-size-extraSmall;
+    width: $star-size-extraSmall * $star-count + $star-spacing;
```
Comment (kevinrodrigues) [inline src/scss/components/_ratings.scss]: Should this calculation be in brackets?

($star-size-extraSmall * $star-count) + $star-spacing

Comment (sonnyprince) [reply src/scss/components/_ratings.scss]: for the calculation, no. for readability, yeah! - Have updated them all 👍

File: src/scss/components/_ratings.scss [hunk 3]
```diff
@@ -1,30 +1,37 @@
...
     @extend %c-icon--star--empty !optional;
     background-repeat: repeat-x;
     background-size: contain;
     display: inline-block;
     height: $star-size-small;
-    width: $star-size-small * $star-count + $star-spacing;
+    width: ($star-size-small * $star-count) + $star-spacing;
+}
+
+.c-rating--extraSmall {
```
Comment (ashleynolan) [inline src/scss/components/_ratings.scss]: Can you add this modifier into the ratings page on the docs site too

File: src/scss/components/_ratings.scss [hunk 4]
```diff
@@ -1,30 +1,37 @@
-$star-count: 6;
-$star-size-small: 16px;
-$star-size-medium: 28px;
-$star-size-large: 42px;
-$star-spacing: $star-count;
+$star-count             : 6;
+$star-size-extraSmall   : 12px;
+$star-size-small        : 16px;
```
Comment (ashleynolan) [inline src/scss/components/_ratings.scss]: Might be best to change `$star-size-small` to `$star-size-default` as that's the default sizing used.

Then the new size can be `$star-size-small` as it's making it smaller than the default (make sense?).

Also, would be good to change the names to be `$star-size--small` (double dash as they're modifiers) if that doesn't mean too much re-work.

File: src/scss/components/_ratings.scss (hunk 3 above)
Comment (sonnyprince) [reply src/scss/components/_ratings.scss]: done - PR is open and linked in the description :)

**Pull request merged**